apps/etterlatte-trygdetid-kafka/src/main/kotlin/MigreringHendelser.kt

```diff
@@ -58,18 +59,40 @@ internal class MigreringHendelser(rapidsConnection: RapidsConnection, private va
                     val request = objectMapper.treeToValue(packet[HENDELSE_DATA_KEY], MigreringRequest::class.java)
                     logger.info("Oppretter grunnlag for trygdetid for $behandlingId")
 
-                    // Oppretter alle trygdetidsperioder og beholder siste beregnede trygdetid
-                    val beregnetTrygdetid =
-                        request.trygdetid.perioder.map { periode ->
-                            trygdetidService.beregnTrygdetidGrunnlag(behandlingId, tilGrunnlag(periode))
-                        }.lastOrNull() ?: it
+                    try {
+                        // Oppretter alle trygdetidsperioder og beholder siste beregnede trygdetid
+                        val beregnetTrygdetid =
+                            request.trygdetid.perioder.map { periode ->
+                                trygdetidService.beregnTrygdetidGrunnlag(behandlingId, tilGrunnlag(periode))
+                            }.lastOrNull() ?: it
 
-                    packet[TRYGDETID_KEY] = beregnetTrygdetid.toJson()
-                    packet.eventName = Migreringshendelser.BEREGN
-                    context.publish(packet.toJson())
-                    logger.info(
-                        "Publiserte oppdatert migreringshendelse fra trygdetid for behandling $behandlingId",
-                    )
+                        packet[TRYGDETID_KEY] = beregnetTrygdetid.toJson()
+                        packet.eventName = Migreringshendelser.BEREGN
+                        context.publish(packet.toJson())
+                        logger.info(
+                            "Publiserte oppdatert migreringshendelse fra trygdetid for behandling $behandlingId",
+                        )
+                    } catch (e: Exception) {
+                        logger.warn("Migrering for behandling $behandlingId feilet - overstyrer trygdetid", e)
+
+                        // TODO - EY-2602 - flere felter trengs - feks utland
+                        val beregnetTrygdetid =
+                            DetaljertBeregnetTrygdetidResultat.fraSamletTrygdetidNorge(
+                                request.beregning.anvendtTrygdetid.toInt(),
+                            )
+
+                        trygdetidService.overstyrBeregnetTrygdetid(
+                            behandlingId,
+                            beregnetTrygdetid,
+                        )
+
+                        packet[TRYGDETID_KEY] = beregnetTrygdetid.toJson()
```

Comment:
Endel felles logikk her for kafkameldingen som sendes for når det går bra og når vi overstyrer. Kunne kanskje dratt det inn i en egen metode.

Reply:
Gjort